RavenDB-15291

aviv86 opened the pull request:
https://issues.hibernatingrhinos.com/issue/RavenDB-15291

aviv86 committed:
RavenDB-15291 : do not recreate new counters storage after each commit -
if the first time that we create 'CounterGroupsKeys' is via 'new CounterStorage()' 'CounterGroupsKeys' tree will have 'IsIndexTree = false', which can cause errors when attempting to delete pages

aviv86 committed:
RavenDB-15291 : after we finished processing all counters of a document - recreate document counters in metadata and remove "@counters-snapshot" if exists

ppekrol commented on src/Raven.Server/Storage/Schema/Updates/Documents/50000/From42017.cs:
```diff
@@ -141,6 +139,86 @@ public bool Update(UpdateStep step)
+                    return; // document doesn't exists 
+                
+                var tableId = tvr.Id;
+                var counterNames = step.DocumentsStorage.CountersStorage.GetCountersForDocumentList(context, step.WriteTx, docId);
```
better to use `GetCountersForDocument()` and do `.ToList()` 

This way we can avoid having `GetCountersForDocumentList` method

aviv86 committed:
RavenDB-15291 : minor changes
- address PR comment - use GetCountersForDocument, remove GetCountersForDocumentList
- dispose lazy strings when we finish batch

ppekrol committed:
RavenDB-15291 after rebase cleanup
